Formula/jpeg-xl.rb

```diff
@@ -0,0 +1,32 @@
+class JpegXl < Formula
+  desc "New file format for still image compression"
+  homepage "https://jpeg.org/jpegxl/index.html"
+  url "https://gitlab.com/wg1/jpeg-xl.git", tag: "v0.2", revision: "31c71b0f61123a40789b0b8f54feb70e5995420e"
```

Comment:
```suggestion
  url "https://gitlab.com/wg1/jpeg-xl.git",
      tag:      "v0.2",
      revision: "31c71b0f61123a40789b0b8f54feb70e5995420e"
```

Can't we use a release tarball instead? https://gitlab.com/wg1/jpeg-xl/-/releases

Reply:
I did attempt building building from the tarball, but wasn't sure on the best way to get the third_party directory populated. With git, it's just submodules.

Reply:
Wait, what goes into the `third_party` directory? What other dependencies does this need?

Reply:
that mostly implies the upstream release tarball issue (which is supposed to package the `third_party` resources as well), as a workaround, we can do what @carlocab said by downloading the resources manually.

Reply:
The gitlab repo specifies commit hashes for its submodules, though, so I think using a git checkout here should be fine?

Reply:
I've had a shot at making a formula which builds from the tarball and uses the resources method to grab the third party dependencies.  This seems to work:

https://gist.github.com/dkam/9f231d9543be0f2e51f9f538155b75d3

Let me know if you'd prefer that formula. If you do, should I modify this pull request or make a new one?

Reply:
BTW - the tarball based formula drops the ```fails_with :clang``` line as it caused one of the third_party dependancies to fail compilation - strangely jpeg-xl seemed to compile ok without it now.

Reply:
Thanks for looking into this. Using the resources looks complicated to update; let's just stick with the gitlab checkout here.

Reply:
Hi, jxl dev here. The tarball is automatically generated by gitlab and it doesn't bundle the submodules (I don't think there's a way to do that in gitlab, please let me know if there is one).

For the core library only google/highway, google/brotli and skiacms are required at the moment, and then for the cjxl/djxl tools we need as well the other image decoders so those tools can convert from PNG/JPEG/GIF/etc to JXL: lodepng, libjpeg, etc, .... (some of these like gif or exr are optional, but the resulting cjxl/djxl would not support that format if not provided obviously). Most of these libraries are already available in many platforms so they normally shouldn't be bundled with jpeg-xl and instead use the ones provided by the system. For example, in this recipe you used `depends_on "giflib"` to make giflib available, but if we had bundled giflib in third_party you wouldn't have used it anyway.

For google/highway in particular we also try to look it up from the system with pkg-config so it could have its own recipe. There was recently a 0.7 release of google/highway. 

There are a few extra submodules in the git repo that are not required for compiling the core lib and cjxl/djxl tools that we have for other use cases (like running metrics, compiling some plugins).

For people who downloaded the .tar.gz and need the minimum list of dependencies we provide this [deps.sh script](https://gitlab.com/wg1/jpeg-xl/-/blob/master/deps.sh) that uses curl to download the right version we are targeting. Maybe you can make use of the SHA values from there for easier upgrade here, they are kept in sync with the git submodule values in our repo. Let us know if there is something else we can do that would work better.

Reply:
Thanks deymo.  Given that, think we should move to the tarball version which uses the ```resource``` system to download the required dependencies.  The Git version ( in this pull request ) has now started grabbing other submodules  such as "IQA-optimization" and "vmaf" which aren't required to build.